proposals/NNNN-reinstate-self-for-accessing-instance-properties-and-functions.md

```diff
@@ -0,0 +1,99 @@
+# Re-instate mandatory self for accessing instance properties and functions  
+
+* Proposal: [SE-NNNN](https://github.com/apple/swift-evolution/blob/master/proposals/NNNN-mandatory-self-for-accessing-instance-properties-and-functions.md)
+* Author(s): [David Hart](https://github.com/hartbit)
+* Status: **Review**
+* Review manager: TBD
+
+## Introduction
+
+Swift used to force using `self.` when accessing instance properties and functions, but since dropped this requirement (only keeping it for closure captures). It now feels inconsistent, and we have lost the automatic documentation of instance properties vs local variables and instance functions vs local functions or closures. This proposal offers to re-instate the original behaviour.
```

Comment:
Surely something like that would have been in the [Patch Notes](https://developer.apple.com/library/prerelease/ios/releasenotes/DeveloperTools/RN-Xcode/Chapters/xc6_release_notes.html#//apple_ref/doc/uid/TP40001051-CH4-SW26) or in the change log on the [Swift blog](https://developer.apple.com/swift/blog/?id=3). Swift 1.2 was released alongside Xcode 6.3 as well. Are you suggesting there was a requirement to use self when reference properties and methods all the way through Swift 1.0-1.1? That simply isn't true. 

Here, look at the slide deck for [Introduction to Swift](http://devstreaming.apple.com/videos/wwdc/2014/402xxgg8o88ulsr/402/402_introduction_to_swift.pdf) at WWDC 2014. Search for the Computed Properties section. This is the first slide shown referencing a property of a class in swift. No self.